portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -53,21 +55,36 @@ private void writeToDatabase(Entity toWrite) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String limitString = getParameter(request, "number-comments", "5");
+    int limitInt = parseToInteger(limitString);
+    
+    if(limitInt > 100) {
+        limitInt = 100;
+    } else if(limitInt <= 0) {
+        limitInt = 1;
+    }
 
-    List<Comment> comments = getComments();
+    List<Comment> comments = getComments(limitInt);
 
     Gson gson = new Gson();
-
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(comments));
   }
 
-  private List<Comment> getComments() {
+  private int parseToInteger(String str) {
+    try {
+      return Integer.parseInt(str);
+    } catch(NumberFormatException exception) {
+      return 5; //default value
```

Comment:
I don't think you want to hardcode 5 here in case parseToInteger is used elsewhere.

consider either returning null or passing the default value as an argument to this function.

Reply:
Good point, I changed it so that the default value is passed as an argument.